mesmer/core/datatree.py

```diff
@@ -70,65 +70,140 @@ def collapse_datatree_into_dataset(
     return ds
 
 
+def stack_datatree(
+    dt: xr.DataTree,
+    *,
+    member_dim: str | None = "member",
+    time_dim: str = "time",
+    scenario_dim: str = "scenario",
+    sample_dim: str = "sample",
+) -> xr.Dataset:
+    """
+    stack datatree along a new dimension, named sample_dim. Each node needs a time- and
+    member-dimension. The scenario dimension from the name of each node.
+
+    Parameters
+    ----------
+    dt : xr.DataTree
+        DataTree to stack
+    member_dim : str | None, default: "member"
+        Name of the member dimension.
+    time_dim : str, default: "time"
+        Name of the time dimension.
+    scenario_dim : str, default: "scenario"
+        Name of the scenario dimension.
```

Comment:
Actually, do we need that? Is the name of the dimension still there after stacking? Yes right?

```suggestion
        Name of the scenario dimension that will used in the stacked dataset.
```